src/bos/operators/base.py

```diff
@@ -122,6 +122,7 @@ def _run(self) -> None:
             components = self._act(components)
         except Exception as e:
             LOGGER.error("An unhandled exception was caught while trying to act on components: {}".format(e))
+            LOGGER.exception(e)
```

Comment:
Why log the error a second time?

Reply:
In this particular case, logging the error message was insufficient for tracking down the exact location in the codebase that yielded it. When we log it as an exception, it additionally gives us the full stack location so that follow-up can occur.

Reply:
You can do this without the extra message by adding "stack_info=True" as an argument to LOGGER.error

Reply:
Or just change the first call to "exception" rather than "error".  I just hate to clutter the logs with duplicate error messages.

Reply:
I had considered the latter, but determined that calling the error exceptional (we didn't or couldn't handle it, except by nature of a blank except Exception python no-no) was still valuable. It means "there is probably a bug here because something we didn't expect to happen, did." As was the case, the message about the unbound local error from existing code when PCS had an undefined power state in existing code caused me to leave the existing message in place. It isn't harmful and it also does a good job of explaining why we're seeing the stack in the first place.

Reply:
It sounds like you're set on this, and it's not important enough to hold this up, but I will just add that "exception" isn't a log level.  LOGGER.exception logs at an ERROR level, and in fact its really just a shortcut for LOGGER.error(exec_info=True), so we're really just logging the same error at the same level twice with this.

Reply:
Yep, you're right -- they funnel to the same function and one is just short hand for the other. I can switch it up.